demos/observers/bolometry/irvb.py

```diff
@@ -0,0 +1,300 @@
+
+# The camera consists of a box with a rectangular slit and foil.
+# In its local coordinate system, the camera's slit is located at the
+# origin and the foil below the z=0 plane, looking up towards the slit.
+
+# To position the camera relative to its parent, set the `transform`
+# property to produce the correct translation and rotation.
+camera_box = Box(lower=Point3D(-BOX_WIDTH / 2, -BOX_HEIGHT / 2, -BOX_DEPTH),
+                 upper=Point3D(BOX_WIDTH / 2, BOX_HEIGHT / 2, 0))
+# Hollow out the box
+outside_box = Box(lower=camera_box.lower - Vector3D(1e-5, 1e-5, 1e-5),
+                  upper=camera_box.upper + Vector3D(1e-5, 1e-5, 1e-5))
+camera_box = Subtract(outside_box, camera_box)
+# The slit is a hole in the box.
+aperture = Box(lower=Point3D(-SLIT_WIDTH / 2, -SLIT_HEIGHT / 2, -1e-4),
+               upper=Point3D(SLIT_WIDTH / 2, SLIT_HEIGHT / 2, 1e-4))
+camera_box = Subtract(camera_box, aperture, name="Camera box geometry")
+camera_box.material = AbsorbingSurface()
+# Instance of the bolometer camera.
+bolometer_camera = BolometerCamera(camera_geometry=camera_box, parent=world,
+                                   name="Demo camera")
+# The bolometer slit in this instance just contains targeting information
+# for the ray tracing, since we have already given our camera a geometry.
+# The slit is defined in the local coordinate system of the camera.
+slit = BolometerSlit(slit_id="Example slit", centre_point=ORIGIN,
+                     basis_x=XAXIS, dx=SLIT_WIDTH, basis_y=YAXIS, dy=SLIT_HEIGHT,
+                     parent=bolometer_camera)
+foil_transform = translate(0, 0, -SLIT_SENSOR_SEPARATION)
+foil = BolometerIRVB(
+    detector_id="IRVB foil", centre_point=ORIGIN.transform(foil_transform),
+    basis_x=XAXIS.transform(foil_transform), basis_y=YAXIS.transform(foil_transform),
+    width=FOIL_WIDTH, pixels=PIXELS, slit=slit, parent=bolometer_camera,
+    units="Power", accumulate=False, curvature_radius=FOIL_CORNER_CURVATURE)
+bolometer_camera.add_foil_detector(foil)
+
+# The camera is positioned at (x, y, z) = (1.5, 2.5, 0)m and looking along the -x axis.
+bolometer_camera.transform = translate(1.5, 2.5, 0) * rotate_basis(-YAXIS, ZAXIS)
+
+# Pick out just the foil, which we'll use for observing later.
+irvb_foil = bolometer_camera[0]
+
+########################################################################
+# Show the lines of sight of this camera.
+########################################################################
+# Set up a representative tokamak geometry.
+eq = example_equilibrium()
+r_inner = eq.limiter_polygon[:, 0].min()
+r_outer = eq.limiter_polygon[:, 0].max()
+z_lower = eq.limiter_polygon[:, 1].min()
+z_upper = eq.limiter_polygon[:, 1].max()
+dr = r_outer - r_inner
+dz = z_upper - z_lower
+outer_wall = Cylinder(radius=3, height=3, transform=translate(0, 0, -1.5))
+first_wall = axisymmetric_mesh_from_polygon(eq.limiter_polygon, 20)
+vessel = Subtract(outer_wall, first_wall)
```

Comment:
The lines below are failing because of the clash between inside / outside the CSG compound. Its hard to see without the Raysect visualisation tools in my private branch.

The outer wall is at a much larger radius than the inner wall. The 'irvb_port' cylinder cuts through the first wall but isn't long enough to also cut through the outer wall. So what is inside / outside?

For this example you don't actually need the outer_wall subtraction. I changed these lines to this and then it worked for me. But you might want to move the camera into the proper position and let me know if it still works.

Try changing 105-107 to simply:
`vessel = axisymmetric_mesh_from_polygon(eq.limiter_polygon, 20)`

Reply:
This worked, thanks. I didn't realise `axisymmetric_mesh_from_polygon` created a hollow mesh: it's not obvious from the code that this is the case.